extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java

```diff
@@ -321,8 +321,9 @@ public Uni<? extends SecurityIdentity> apply(Throwable t) {
 
                                             if (!expired) {
                                                 LOG.errorf("ID token verification failure: %s", t.getCause());
-                                                return Uni.createFrom()
-                                                        .failure(new AuthenticationCompletionException(t.getCause()));
+                                                return removeSessionCookie(context, configContext.oidcConfig)
+                                                        .replaceWith(Uni.createFrom()
+                                                                .failure(new AuthenticationCompletionException(t.getCause())));
```

Comment:
Check if `t instanceof AuthenticationCompletionException`?

Reply:
@gastaldi Sure, I think `OidcIdentityProvider` would not throw it, but just in case, make sense, let me do it now